pkg/client/unversioned/transport.go

```diff
@@ -171,30 +171,43 @@ func NewClientCertTLSConfig(certData, keyData, caData []byte) (*tls.Config, erro
 	if err != nil {
 		return nil, err
 	}
-	certPool := x509.NewCertPool()
-	certPool.AppendCertsFromPEM(caData)
+
+	caPool := getCertPoolForCA(caData)
+
 	return &tls.Config{
 		// Change default from SSLv3 to TLSv1.0 (because of POODLE vulnerability)
 		MinVersion: tls.VersionTLS10,
 		Certificates: []tls.Certificate{
 			cert,
 		},
-		RootCAs:    certPool,
-		ClientCAs:  certPool,
+		RootCAs:    caPool,
+		ClientCAs:  caPool,
 		ClientAuth: tls.RequireAndVerifyClientCert,
 	}, nil
 }
 
 func NewTLSConfig(caData []byte) (*tls.Config, error) {
-	certPool := x509.NewCertPool()
-	certPool.AppendCertsFromPEM(caData)
 	return &tls.Config{
 		// Change default from SSLv3 to TLSv1.0 (because of POODLE vulnerability)
 		MinVersion: tls.VersionTLS10,
-		RootCAs:    certPool,
+		RootCAs:    getCertPoolForCA(caData),
 	}, nil
 }
 
+func getCertPoolForCA(caData []byte) *x509.CertPool {
+	// What we really want is a copy of x509.systemRootsPool, but that isn't exposed.  It's difficult to built (see the go
+	// code for a look at the platform specific insanity), so we'll use the fact that RootCAs == nil gives us the system values
+	// It doesn't allow trusting either/or, but hopefully that won't be an issue
+	if len(caData) == 0 {
```

Comment:
> What is this set to if a CA file is specified which is empty? Wondering if == nil would be a better check

I would expect the system check to be strictly better than nothing.  Otherwise, why have it be the system check?  If I had a way to do it, I would return the union of the two.

It seems that in either the empty or missing case, the information is unavailable and we'd want to fallback to the system check which should never contain a signer that you don't trust.  Its "failing" in the safe direction.